### PR TITLE
Mention more tools in tool-recommendations.rst

### PR DESCRIPTION
**Pull request by antnieszka**
Hi,

I think it's worth mentioning in the docs, that we have more tools available. As this is my first contribution here, any suggestions would be greatly appreciated. Just wanted to start with something.

Motivation - pipenv is not the only option out there and seems a bit abandoned:
https://github.com/pypa/pipenv/issues/4024
https://github.com/pypa/pipenv/issues/4058

### CODE REVIEW

**Comment by ncoghlan** [comment]
While I expect Dan to be able to get pipenv back on track in the new year, pip-tools and poetry are still worth highlighting here, since pipenv doesn't aim to be all things to all people, instead leaving some workflows to these other options.

File: source/guides/tool-recommendations.rst [hunk 1]
```diff
@@ -15,8 +15,11 @@ Use :ref:`pipenv` to manage library dependencies when developing Python
 applications. See :doc:`../tutorials/managing-dependencies` for more details
 on using ``pipenv``.
 
-Consider other tools such as :ref:`pip` when ``pipenv`` does not meet your use
-case.
+When :ref:`pip` and ``pipenv`` do not meet your use case consider other tools:
```
[the remark this answers is not in the document]
Comment (antnieszka) [reply source/guides/tool-recommendations.rst]: Sure, should I add "user pip directly" as third bullet point?

[the remark this answers is not in the document]
Comment (pradyunsg) [reply source/guides/tool-recommendations.rst]: Rather, as the first. :)

**Comment by zekefarwell** [comment]
Please do make this change.  As a new user of Python in recent months, I was confused by [this section on packaging.python.org](https://packaging.python.org/guides/tool-recommendations/#application-dependency-management).  It made it seem like pipenv was intended to become the new official package manager for python.  After a lot of reading, I now understand that is not the case.  I think changing the wording here will make things clearer to new python users.

In fact I'd rewrite the whole section like this:

> ## Application dependency management
> There are several options for managing library dependencies when developing Python applications.  See Managing Application Dependencies for more details.
> * __Pip__  - The default package manager included with Python.  It handles a small number of dependencies just fine, but for complex dependency management other options are recommended.
> * __Pip-tools__ - A lightweight wrapper for pip that adds a few helpful features.
> * __Pipenv__ - A full featured dependency manager, similar in spirit to Node.js’ npm or Ruby’s bundler.  A higher-level tool than pip, it simplifies dependency management for common use cases.
> * __Poetry__ - A newer tool, comparable in scope to pipenv.  In addition to dependency management it handles building and packaging of projects for distribution. 

The [Managing Application Dependencies](https://packaging.python.org/tutorials/managing-dependencies/) page also contributed to my confusion about `pipenv` seeming like the new official python package manager.  I think it should be titled "Managing Application Dependencies With Pipenv" or expanded to give equal weight to pip, pip-tools, pipenv, and poetry.

**Comment by ncoghlan** [comment]
We really do want people to try pipenv first, and then only explore the other options if pipenv isn't working out for them.

(pip & pip-tools require too much custom scripting to be the default recommendation for app dependencies, while poetry is focused more on developing publishing new Python components rather than apps that happen to be written in Python)

**Comment by zekefarwell** [comment]
I understand.  I just wanted to share my perspective as someone recently starting out in Python.  I did try pipenv first, based on the recommendation of [packaging.python.org](https://packaging.python.org).  However, all the documentation on [docs.python.org](https://docs.python.org) described using pip with no mention of pipenv.  As a new user, this mixed message across python.org was confusing.  Things would have been a lot clearer to me if these pages had explained the various options for package management rather than simply recommending pipenv.

**Comment by ncoghlan** [comment]
Which other documentation seemed confusing? We've generally been trying to reduce the coverage of packaging tools outside packaging.python.org, and instead have it redirect to the main packaging docs, so if there are other pages that still appear to be presenting themselves as authorative, those are the pages that should be reworded.

**Comment by zekefarwell** [comment]
It wasn't a specific piece of documentation that was confusing, it was the fact that recommendations across python.org were not consistent.  While packaging.python.org clearly recommends pipenv as the default, docs.python.org recommends pip and makes no mention of pipenv (at least that I saw).  For example, [This tutorial on docs.python.org](https://docs.python.org/3/tutorial/venv.html?highlight=pip%20requirements%20txt#managing-packages-with-pip) describes a pip based workflow and makes no mention of pipenv.  [This one on packaging.python.org](https://packaging.python.org/tutorials/managing-dependencies/) describes a pipenv workflow with some brief mentions of other options.

**Comment by webknjaz** [comment]
@zekefarwell packaging.python.org is a community resource maintained by a group of people who call themselves PyPA while docs.python.org is maintained by the Core Developers of CPython. "docs" only documents what ships along with the standard distribution of CPython and PyPA guides are there to guide you throughout more advanced things.

**Comment by sub-mod** [comment]
> We really do want people to try pipenv first, and then only explore the other options if pipenv isn't working out for them.
> 
Is this the PyPA recommendation ?
I still see a lot of people using pip as  package manager for python applications. And there seems to be a resistance among cloud container providers from using pipenv in their containers.None of the containers I saw in the cloud had pipenv.

**Comment by zekefarwell** [comment]
@webknjaz, I understand all of that now, having read many blog posts and comment threads.  I'm trying to provide the perspective of a newcomer to the Python ecosystem since I went through that a few months ago.  The newcomer experience starts at python.org, and docs.python.org, where pip is recommended.  At some point you hear about pipenv and a google search turns up the tutorials on packaging.python.org.  Though I now understand why these subdomains document different things, the newcomer experience is weird.  You don't understand the inner workings of the python community, and it just seems like there are multiple, conflicting, official recommendations coming out of python.org.

**Comment by ncoghlan** [comment]
pip is the lowest common denominator, and likely all you need if you're a solo developer working on a single machine (especially for ad hoc scripting use cases).

The recommendation for pipenv only comes in once you're working on an application, and need to provide automation for routinely setting up new development environments.

So if folks are using pip directly and not finding it frustrating in any way, they presumably don't have the problems the more complex app dependency management tools solve.

**Comment by ncoghlan** [comment]
Note that this is why the packaging.python.org tutorials are structured the way they are:

* first the basics of installing individual packages with pip
* then managing pinned dependencies in a collaborative project with pipenv

**Pull request merged**